src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs

```diff
@@ -184,7 +184,7 @@ internal AveragedPerceptronTrainer(IHostEnvironment env, Options options)
                 DecreaseLearningRate = decreaseLearningRate,
                 L2Regularization = l2Regularization,
                 NumberOfIterations = numberOfIterations,
-                LossFunction = lossFunction ?? new HingeLoss()
```

Comment:
Doesn't this remove the `HingeLoss` default?

Reply:
it will default by `LossFunctionFactory`

[https://github.com/dotnet/machinelearning/blob/a3d38135115f0f0a67b20b490f0871665e9502a4/src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs#L156C36-L156C36](https://github.com/dotnet/machinelearning/blob/a3d38135115f0f0a67b20b490f0871665e9502a4/src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs#L156C36-L156C36)

[https://github.com/dotnet/machinelearning/blob/a3d38135115f0f0a67b20b490f0871665e9502a4/src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs#L119C71-L119C71](https://github.com/dotnet/machinelearning/blob/a3d38135115f0f0a67b20b490f0871665e9502a4/src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs#L119C71-L119C71)

[https://github.com/dotnet/machinelearning/blob/a3d38135115f0f0a67b20b490f0871665e9502a4/src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs#L100](https://github.com/dotnet/machinelearning/blob/a3d38135115f0f0a67b20b490f0871665e9502a4/src/Microsoft.ML.StandardTrainers/Standard/Online/AveragedPerceptron.cs#L100)